CRM/Contact/BAO/Query.php

```diff
@@ -4521,6 +4521,59 @@ public static function filterCountryFromValuesIfStateExists(&$formValues) {
   }
 
   /**
+   * Include Select columns in groupBy clause.
+   *
+   * @param array $selectClauses
+   * @param array $groupBy - Columns already included in GROUP By clause.
+   *
+   * @return string
+   */
+  public static function getGroupByFromSelectColumns($selectClauses, $groupBy = NULL) {
+    $groupBy = (array) $groupBy;
+    $mysqlVersion = CRM_Core_DAO::singleValueQuery('SELECT VERSION()');
+    $sqlMode = CRM_Core_DAO::singleValueQuery('SELECT @@sql_mode');
+
+    //return if ONLY_FULL_GROUP_BY is not enabled.
+    if (!version_compare($mysqlVersion, '5.7', '<') && !empty($sqlMode) && in_array('ONLY_FULL_GROUP_BY', explode(',', $sqlMode))) {
```

Comment:
There are a few reason behind this-

1) Main reason being the `ANY_VALUE` keyword which is only introduced in 5.7+ which we use it here in some summary reports (Membership, Contribution, etc) which enforces the result to group by selected columns.

2) MySQL 5.7+ has some enhancements to `ONLY_FULL_GROUP_BY` mode that recognises Functional Dependencies unlike in lower versions which work on blindly including all the select columns in `group_by` clause.